saleor/payment/utils.py

```diff
@@ -68,23 +73,18 @@ def create_payment(**payment_data):
     return payment
 
 
-def create_transaction(
-        payment: Payment,
-        token: str,
-        kind: str,
-        is_success: bool,
-        amount: Decimal,
-        currency: str,
-        gateway_response: Optional[Dict] = None) -> Transaction:
-    if not gateway_response:
+def create_transaction(payment: Payment, kind: str, payment_token: str,
+        gateway_response: Dict=None, **extra_params) -> Transaction:
+    """Creates a Transaction based on transaction kind and gateway response."""
+    if gateway_response is None:
         gateway_response = {}
     txn, _ = Transaction.objects.get_or_create(
         payment=payment,
-        token=token,
         kind=kind,
-        is_success=is_success,
-        amount=amount,
-        currency=currency,
+        token=gateway_response.get('transaction_id', payment_token),
+        is_success=gateway_response.get('is_success', False),
+        amount=extra_params.get('amount', payment.total),
+        currency=payment.currency,
         gateway_response=gateway_response)
```

Comment:
See my other comment, doesn't really matter what the key is named, the whole response is saved. Whether the gateway wants to send one is up to the gateway :P

Reply:
Yeah, but it will be nice to keep it as a convention. You are using ```raw_response``` in braintree but ```gateway_response``` in dummy. Why not just keep them same as ```raw_response```.

Reply:
In this case it is better removing ```gateway_response``` from dummy, otherwise it is confusing. And the docs should have explanations for this gateway response.